wiring/inc/spark_wiring_system.h

```diff
@@ -148,8 +148,18 @@ class SystemClass {
 
     inline int enableFeature(HAL_Feature feature)
     {
+        /* PRE-ENABLE TASKS */
+        bool init_sram = false;
+        // Initialize SRAM on first use of retained memory
+        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_RETAINED_MEMORY)) {
+            init_sram = true;
+        }
+
+        /* ENABLE FEATURE */
         int result = HAL_Feature_Set(feature, true);
-        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_WARM_START)) {
+
+        /* POST-ENABLE TASKS */
+        if (feature==FEATURE_RETAINED_MEMORY && init_sram) {
```

Comment:
We are already initializing about 50 KBytes of regular RAM, so another 4K is a very small increase. I imagine we are talking sub millisecond times here, and nothing compared to the default wait of 1 second for the USB bus to initialize.   Also only the data that is used is initialized and not the entire 4KB.